hollow/src/test/java/com/netflix/hollow/core/util/HashCodesTest.java

```diff
@@ -41,7 +45,91 @@ public void testStringHashCode() {
             Assert.assertEquals(accurateStringHashCode(str), HashCodes.hashCode(str));
         }
     }
-    
+
+    @Test
+    public void testHashTableSize() {
+        // Current load factor is 10 / 7. If load factor calculation is changed, this test should be updated
+        int N;
+
+        try {
+            HashCodes.hashTableSize(-1);
+            Assert.fail("exception expected");
+        } catch (IllegalArgumentException ex) {
+            Assert.assertEquals("cannot be negative; numElements=-1", ex.getMessage());
+        }
+
+        Assert.assertEquals(1, HashCodes.hashTableSize(0));
+        Assert.assertEquals(2, HashCodes.hashTableSize(1));
+        Assert.assertEquals(4, HashCodes.hashTableSize(2));
+
+        // first integer overflow boundary condition (214_748_364)
+        N = Integer.MAX_VALUE / 10;
+        Assert.assertEquals(1 << 29, HashCodes.hashTableSize(N));
+        Assert.assertEquals(1 << 29, HashCodes.hashTableSize(N + 1));
+
+        // first quantity that requires hash table size at maximum
+        N = 375_809_640;
+        Assert.assertEquals(1 << 29, HashCodes.hashTableSize(N - 1));
+        Assert.assertEquals(HASH_TABLE_MAX_SIZE, HashCodes.hashTableSize(N));
+
+        // first quantity (after load factor) that exhausts maximum hash table size
+        N = 751_619_278;
```

Comment:
I have a TODO to add a "Known Limitations" section in [hollow.how](http://hollow.how). This number will factor into a discussion on the upper limits of cardinality (if I understand correctly) of primary keys, sets, and maps.